Add headers to all markdown files (#4627)
ethereum/ethereum-org-website: src/content/developers/tutorials/logging-events-smart-contracts/index.md

address: "0x19dE91Af973F404EDF5B4c093983a7c6E3EC8ccE"
---

> # Logging data from smart contracts with events {#logging-data-from-smart-contracts-with-events}

Reviewer: Why do we need to add a header to the beginning of tutorials? (Is the lack thereof causing issues with the translation support?) If it is possible I'd much rather remove leading headers elsewhere rather than adding them here.

If we do need to add headers we should probably use H2s as [we already have a H1 on all tutorial pages](https://github.com/ethereum/ethereum-org-website/blob/994a03c747ced8022b7064050fd68b83fa670381/src/templates/tutorial.js#L177) that we pull from the title in the frontmatter.

> # Logging data from smart contracts with events {#logging-data-from-smart-contracts-with-events}
> 
> In Solidity, [events](/developers/docs/smart-contracts/anatomy/#events-and-logs) are dispatched signals the smart contracts can fire. DApps, or anything connected to Ethereum JSON-RPC API, can listen to these events and act accordingly. Event can also be indexed, so that the event history is searchable later.
> 
> The most common event on the Ethereum blockchain at the time of writing this article is the Transfer event that is emitted by ERC20 tokens when someone transfer tokens.

Reviewer: ```suggestion
In Solidity, [events](/developers/docs/smart-contracts/anatomy/#events-and-logs) are dispatched signals the smart contracts can fire. DApps, or anything connected to Ethereum JSON-RPC API, can listen to these events and act accordingly. An event can also be indexed so that the event history is searchable later.

## Events {#events}

The most common event on the Ethereum blockchain at the time of writing this article is the Transfer event that is emitted by ERC20 tokens when someone transfers tokens.
```
